onnxruntime/contrib_ops/cuda/math/gemm_float8.cc

```diff
@@ -14,17 +14,23 @@ namespace onnxruntime {
 namespace contrib {
 namespace cuda {
 
-#define REGISTER_KERNEL()                                                                                          \
-  ONNX_OPERATOR_KERNEL_EX(                                                                                         \
-      GemmFloat8,                                                                                                  \
-      kMSDomain,                                                                                                   \
-      1,                                                                                                           \
-      kCudaExecutionProvider,                                                                                      \
-      (*KernelDefBuilder::Create())                                                                                \
-          .TypeConstraint("TA", BuildKernelDefConstraints<Float8E4M3FN, Float8E5M2, MLFloat16, BFloat16, float>()) \
-          .TypeConstraint("TB", BuildKernelDefConstraints<Float8E4M3FN, Float8E5M2, MLFloat16, BFloat16, float>()) \
-          .TypeConstraint("TR", BuildKernelDefConstraints<Float8E4M3FN, Float8E5M2, MLFloat16, BFloat16, float>()) \
-          .TypeConstraint("TS", BuildKernelDefConstraints<float>()),                                               \
+#if !defined(DISABLE_FLOAT8_TYPES)
+#define GEMM_FLOAT8_CONSTRAINTS BuildKernelDefConstraints<Float8E4M3FN, Float8E5M2, MLFloat16, BFloat16, float>()
```

Comment:
Can we add a static assert here to ensure that the CUDA version is >= 11.8 ? If it isn't, can we provide the user with a hint that they could build with float8 types disabled since their CUDA version is < 11.8 ? Otherwise, the user possibly will not be aware that such an option exists unless they start looking for relevant documentation.

Reply:
I added this verification in build.py. If CUDA version is known and < 11.8, it raises an exception.

Reply:
Are users required to provide cuda version while building using build.py ? I would have thought it's much more reliably available via the `CUDA_VERSION` macro ?

Reply:
There is no `#pragma error` in the code. That's why I did not add one. The compilation should fail though, symbol `CUDA_R_8F_E4M3` should not be found.

Reply:
done